src/eviction/eviction.c

```diff
@@ -60,7 +59,8 @@ static inline uint32_t ocf_evict_part_do(struct ocf_request *req,
 		return 0;
 	}
 
-	return ocf_eviction_need_space(req->cache, req, target_part, to_evict);
+	return ocf_request_space(req->cache, req, target_part->runtime,
+		target_part->id, to_evict);
```

Comment:
Maybe we can just have an `id` field inside the `runtime` to avoid passing this additional argument everywhere. Seems that it's strongly associated anyway.

Reply:
added ocf_part structure to store runtime pointer as well as id for both user partitions and freelist